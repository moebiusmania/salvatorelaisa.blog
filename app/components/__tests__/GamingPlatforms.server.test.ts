import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import GamingPlatforms from "../GamingPlatforms.server.vue";

describe("GamingPlatforms", () => {
  const mockPlatforms = [
    {
      label: "PlayStation 5",
      url: "https://example.com/ps5",
    },
    {
      label: "Nintendo Switch",
      url: "https://example.com/switch",
    },
  ];

  // Snapshot test
  it("renders correctly", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: mockPlatforms,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders 'Giocato su:' text", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: mockPlatforms,
      },
    });
    expect(wrapper.text()).toContain("Giocato su:");
  });

  it("renders all platform links", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: mockPlatforms,
      },
    });

    const links = wrapper.findAll("a");
    expect(links.length).toBe(mockPlatforms.length);
  });

  it("renders correct platform labels and URLs", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: mockPlatforms,
      },
    });

    const links = wrapper.findAll("a");
    links.forEach((link, index) => {
      expect(link.text()).toBe(mockPlatforms[index].label);
      expect(link.attributes("href")).toBe(mockPlatforms[index].url);
    });
  });

  it("applies sl-badge class to links", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: mockPlatforms,
      },
    });

    const links = wrapper.findAll("a");
    links.forEach((link) => {
      expect(link.classes()).toContain("sl-badge");
    });
  });

  it("handles empty platforms array", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: [],
      },
    });

    expect(wrapper.text()).toContain("Giocato su:");
    expect(wrapper.findAll("a").length).toBe(0);
  });

  it("handles undefined platforms prop", () => {
    const wrapper = mount(GamingPlatforms);

    expect(wrapper.text()).toContain("Giocato su:");
    expect(wrapper.findAll("a").length).toBe(0);
  });

  it("renders platforms in correct order", () => {
    const wrapper = mount(GamingPlatforms, {
      props: {
        platforms: mockPlatforms,
      },
    });

    const links = wrapper.findAll("a");
    expect(links[0].text()).toBe("PlayStation 5");
    expect(links[1].text()).toBe("Nintendo Switch");
  });
});
