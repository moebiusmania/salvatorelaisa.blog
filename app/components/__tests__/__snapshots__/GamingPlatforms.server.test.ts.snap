// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GamingPlatforms > renders correctly 1`] = `"<p data-v-a98cfd4d="" class="platforms"> Giocato su: <a data-v-a98cfd4d="" href="https://example.com/ps5" rel="noopener noreferrer" class="sl-badge">PlayStation 5</a><a data-v-a98cfd4d="" href="https://example.com/switch" rel="noopener noreferrer" class="sl-badge">Nintendo Switch</a></p>"`;
